src/vm.rs

```diff
@@ -533,6 +533,44 @@ impl<'a, T: GuestStagePagingMode> FinalizedVm<'a, T> {
         let mut active_vcpu = vcpu
             .activate(self.vm_pages(), host_context)
             .map_err(|_| EcallError::Sbi(SbiError::InvalidParam))?;
+
+        // Complete a possible pending operation
+        active_vcpu.try_complete_pending_op(|sbi_msg, sbi_ret| -> EcallResult<()> {
```

Comment:
maybe hoist the contents of the closure into it's own function?

Reply:
This will be cleaner yes.